src/java.base/share/classes/java/net/Inet4Address.java

```diff
@@ -134,6 +136,61 @@ class Inet4Address extends InetAddress {
         holder().originalHostName = hostName;
     }
 
+    /**
+     * Creates an {@code Inet4Address} based on the provided IPv4 address literal.
+     * <p> IPv4 address literals in dotted-decimal form and
+     * <a href="Inet6Address.html#format">IPv4-compatible IPv6 address</a> literals
+     * are supported. For example, the following literals are valid literals in
+     * dotted-decimal form:
+     *
+     *   <blockquote><ul style="list-style-type:none">
+     *   <li>{@code 1.2.3.4}</li>
+     *   <li>{@code 06.07.08.09}</li>
```

Comment:
>      *     Inet4Address.ofLiteral("06.07.08.09")        ==>    /6.7.8.9

if we ignore octal encoding, we should document it explicitely.

Maybe 

> The loose IPv4 syntax of partial numbers or hex encoding ("10.0.258", "0xA000102") is not supported and causes a parsing error. Octal (zero) prefixes are ignored, i.e. interpreted decimal.

(However I would prefer to reject octal in this case)

Reply:
This method behaves like `InetAddress.getByName` in that regards.

Reply:
I believe accepting leading 0 as an octal indicator would require a new API, yet to be defined. What we're proposing here is a method that behaves like `InetAddress.getByName` would except that it rejects what `InetAddress.getByName` would have identified as a hostname needing resolution. In other words it only accepts IP literals that `InetAddress.getByName` would accept as IP literals, and parses them as `InetAddress.getByName` would parse them, and rejects anything that would cause `InetAddress.getByName` to invoke the resolver.

Reply:
> if we ignore octal encoding, we should document it explicitely.

Agree - Javadoc for `Inet4Address.ofLiteral` needs to be updated to cleary mention that only IPv4 literals in decimal encoding are supported, and to state that no other encodings (octal, hexadecimal) are supported.  That will also need to mention that one, two, three and four-part IPv4 address literals are supported to match [Inet4Address format section](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/net/Inet4Address.html#format)

> Maybe
>
> > The loose IPv4 syntax of partial numbers or hex encoding ("10.0.258", "0xA000102") is > > not supported and causes a parsing error. Octal (zero) prefixes are ignored, i.e. interpreted decimal.

Thanks for the suggestion. `10.0.258` is supported and [documented](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/net/Inet4Address.html#format) 'd.d.d' format. It is not planned to change support for such address.